tests/plugins/easyimage/manual/progressbarundo.md

```diff
@@ -0,0 +1,14 @@
+@bender-tags: 4.9.0, feature, 932
+@bender-ui: collapsed
+@bender-ckeditor-plugins: sourcearea, wysiwygarea, floatingspace, toolbar, easyimage, undo, resize
+
+## Easy Image Progressbar Undo
+
+1. Upload some images (using drag and drop, copy/paste, etc).
+2. During image upload manipulate selection (click somewhere, select some editor content, etc).
+
+### Expected
+
+**Saved undo steps** counter should not increment on selection manipulation.
+
+**Important**: Selecting or deselecting widget will create an undo step.
```

Comment:
Honestly I don't understand these instructions.

First of all, upload is too quick to allow sensible "selection manipulation". Maybe mocking it with slowed progress would be a good idea?

What's more, how to check if counter does not change, when every selection change will change it?

Reply:
Slowing down an upload makes sense, as network speed throttling is only available on Chrome (and probably you will need 10mb image to test it properly without it).

As for selection change adding snapshot, I'm not sure, tested it on Chrome, FF, Safari, IE11 and it seems pure selection changes does not cause new undo step:

![feb-08-2018 11-40-48](https://user-images.githubusercontent.com/1061942/35968827-43211200-0cc5-11e8-8b63-1c1d21953232.gif)

Maybe there are some particular cases in which id does and some in which it doesn't?

Reply:
There is a case with selecting/deselecting widget which triggers undo snapshot, but it is already covered in tests description.

Reply:
There are also other cases, like select all. Relaying on undo snapshots count may be misleading. I redesigned the test to check if latest created snapshot contains progressbar.